pkg/irsa/capa/capa.go

```diff
@@ -282,6 +274,14 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 	}
 
+	privateKey, err := s.ServiceAccountSecret(ctx)
+	if apierrors.IsNotFound(err) {
+		s.Scope.Logger().Info("Service account is not ready yet, waiting ...")
+		return nil
+	} else if err != nil {
```

Comment:
nit and also a pet peeve of mine. No need to do an else if you're going to return in both cases :D

Reply:
The "waiting" part here is quite confusing, since it's the caller which waits like so

```
// Re-run regularly to ensure OIDC certificate thumbprints are up to date (see `EnsureOIDCProviders`)
return ctrl.Result{RequeueAfter: time.Minute * 5}, nil
```

I can change this little part and add a comment. Will then merge.